transformer_lens/FactoredMatrix.py

```diff
@@ -41,9 +40,9 @@ def __matmul__(
         other: Union[
             Float[torch.Tensor, "... rdim new_rdim"],
             Float[torch.Tensor, "rdim"],
-            FactoredMatrix,
+            "FactoredMatrix",
```

Comment:
Is there a reason for this switch? My understanding is Python would treat this as a string literal (i.e. the type is a string that exactly matches "FactoredMatrix".

Reply:
The switch was made only because beartype worked with one but not the other. The string typehint is actually treated as a forward reference (see https://peps.python.org/pep-0484/#forward-references)

Reply:
Ah cool thanks for the explanation!